google-cloud-datastore/CONTRIBUTING.md

```diff
@@ -0,0 +1,209 @@
+[mocks](https://github.com/seattlerb/minitest#mocks) and
+[minitest-autotest](https://github.com/seattlerb/minitest-autotest).
+
+To run the Datastore unit tests:
+
+``` sh
+$ cd google-cloud-datastore/
+$ bundle exec rake test
+```
+
+### Datastore Documentation Tests
+
+The project tests the code examples in the gem's
+[YARD](https://github.com/lsegal/yard)-based documentation.
+
+The example testing functions in a way that is very similar to unit testing, and
+in fact the library providing it,
+[yard-doctest](https://github.com/p0deje/yard-doctest), is based on the
+project's unit test library, [minitest](https://github.com/seattlerb/minitest).
+
+To run the Datastore documentation tests:
+
+``` sh
+$ cd google-cloud-datastore/
+$ bundle exec rake doctest
+```
+
+If you add, remove or modify documentation examples when working on a pull
+request, you may need to update the setup for the tests. The stubs and mocks
+required to run the tests are located in `support/doctest_helper.rb`. Please
+note that much of the setup is matched by the title of the
+[`@example`](http://www.rubydoc.info/gems/yard/file/docs/Tags.md#example) tag.
+If you alter an example's title, you may encounter breaking tests.
+
+### Datastore Acceptance Tests
+
+The Datastore acceptance tests interact with the live service API. Follow the
+instructions in the {file:AUTHENTICATION.md Authentication guide} for enabling
+the Datastore API. Occasionally, some API features may not yet be generally
+available, making it difficult for some contributors to successfully run the
+entire acceptance test suite. However, please ensure that you do successfully
+run acceptance tests for any code areas covered by your pull request.
+
+To run the acceptance tests, first create and configure a project in the Google
+Developers Console, as described in the {file:AUTHENTICATION.md Authentication
+guide}. Be sure to download the JSON KEY file. Make note of the PROJECT_ID and
+the KEYFILE location on your system.
+
+Before you can run the Datastore acceptance tests, you must first create indexes
+used in the tests.
+
+#### Create Datastore indexes
+
+Install the [gcloud command-line
+tool](https://developers.google.com/cloud/sdk/gcloud/) and use it to create the
+indexes used in the datastore acceptance tests. From the project's root
+directory:
+
+``` sh
+# Install the app component
+$ gcloud components update app
+
+# Set the default project in your env
+$ gcloud config set project PROJECT_ID
+
+# Authenticate the gcloud tool with your account
+$ gcloud auth login
+
+# Create the indexes
+$ gcloud preview datastore create-indexes acceptance/data/
+```
+
+#### Running the Datastore acceptance tests
+
+To run the Datastore acceptance tests:
+
+``` sh
+$ cd google-cloud-datastore/
+$ bundle exec rake acceptance[\\{my-project-id},\\{/path/to/keyfile.json}]
+```
+
+Or, if you prefer you can store the values in the `GCLOUD_TEST_PROJECT` and
+`GCLOUD_TEST_KEYFILE` environment variables:
```

Comment:
Replace all `GCLOUD_` with `GOOGLE_CLOUD_`?

Reply:
The test env vars all begin with `GCLOUD_TEST_`.

Reply:
Oh yeah...